Bugfix: power0 if power_lock is used

stefanbode opened the pull request:
## Description:
power0 changed the power regardless if power_lock was set.


**Related issue (if applicable):** fixes #<Tasmota issue number goes here>

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] The code change is tested and works with Tasmota core ESP8266 V.2.7.6
  - [x] The code change is tested and works with Tasmota core ESP32 V.2.0.14
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

stefanbode committed:
Bug Fix: power_lock with Power0 command

Power0 changed power regardless of the new lock state of power. New implementation respect the LOCK also on global change

stefanbode committed:
simplify

stefanbode commented:
@arendst : Was there a reason to set only on the devices_present the power to 1 and not to the complete 32bit array?. 
So is it required to do:
power_t all_on = POWER_MASK >> (POWER_SIZE - TasmotaGlobal.devices_present);

arendst commented:
> Was there a reason to set only on the devices_present the power to 1 and not to the complete 32bit array?.

Yes. As the print output will show all bits set even if there are just only 1 or two relays/lights present.

arendst commented on tasmota/tasmota_support/support_tasmota.ino:
```diff
@@ -392,13 +392,17 @@ void SetAllPower(uint32_t state, uint32_t source)
-      TasmotaGlobal.power = 0;
-      break;
+        // keep loocked bits and set all other to 0
+        TasmotaGlobal.power &= Settings->power_lock; 
```
This will turn on relays which have locked bit set but were OFF. Not good.

Ditto for the other changes.

stefanbode replied:
ok readded the all_on filter and do some testing. Looks reasonable now to me

stefanbode replied:
Regarding: TasmotaGlobal.power &= Settings->power_lock;  Your finding is wrong. it is a &= and therefore keep only the locked relays ON. All other go OFF as intended.

stefanbode committed:
readded the all_on filter to power_lock change

readded the all_on to ensure new defined relays always start with power off ass intended.

arendst commented:
Thx. Tested. OK
